src/isomorphic/React.js

```diff
@@ -86,4 +86,31 @@ var React = {
 
 };
 
+if (__DEV__) {
+  // React.DOM factories are deprecated. Set up a Proxy to listen for
+  // accesses to the React.DOM namespace and alert users to switch to
+  // the `react-addons-dom-factories` package.
+  if (typeof Proxy === 'function') {
+    var hasWarnedOfFactories = false;
```

Comment:
We could simply it further by using a `for ... in` loop over `ReactDOMFactories`

```js
if (__DEV__) {
  var warnedForFactories = false;
  for (var factory in ReactDOMFactories) {
    React.DOM[factory] = function(...args) {
      if (!warnedForFactories) {
        warning(
          false,
          'Accessing factories like React.DOM.%s has been deprecated ' +
          'and will be removed in the future. Use the ' +
          'react-addons-dom-factories package instead.',
          factory
        )
        warnedForFactories = true;
      }
      return ReactDOMFactories[factory](...args);
    }
  }
}
```